### PR TITLE
JAMES-2111 Remove Cassandra message DAO V1

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/Cid.java [hunk 1]
```diff
@@ -47,6 +48,16 @@ private static String unwrap(String cidAsString) {
         return unwrapCid;
     }
 
+    public static Optional<Cid> fromRelaxedNoUnwrap(String cidAsString) {
```
Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/Cid.java]: I'm trying to understand what this function does from its name and I must say I can't.
I would argue it requires a "policy" like, for example, guava Splitter

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java [hunk 2]
```diff
@@ -33,9 +33,9 @@
 import com.google.common.base.Preconditions;
 
 public class CassandraSchemaVersionManager {
-    public static final int MIN_VERSION = 1;
+    public static final int MIN_VERSION = 2;
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java]: how is it related to JAMES-2096 ?

Comment (chibenwa) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java]: It indicates code do not support anymore schemes version 1 and dropped tge tables

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/Cid.java (hunk 1 above)
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/Cid.java]: This comment is unrelated to this PR and deserve to be here: https://github.com/linagora/james-project/pull/921

(However, it is important to not remove table migration before being able to migrate ALL old entries, hence I put this branch on top of the other one to not forget nor do mistakes while merging.)

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java (hunk 2 above)
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java]: but not the goal of the ticket 
```
Expose migration tool written in JAMES-2083 over webadmin.
Ensure concurrent run can not be made.
```

Comment (chibenwa) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionManager.java]: I'll create a new ticket then.

**Comment by chibenwa** [comment]
https://issues.apache.org/jira/browse/JAMES-2111

File: server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java [hunk 3]
```diff
@@ -75,7 +74,15 @@
...
     public void setUp() throws Exception {
         guiceJamesServer = cassandraJmapTestRule.jmapServer()
-                .overrideWith(new WebAdminConfigurationModule());
+                .overrideWith(new WebAdminConfigurationModule(),
+                    new AbstractModule() {
+                        @Override
+                        protected void configure() {
+                            MapBinder<Integer, Migration> mapBinder = MapBinder.newMapBinder(binder(), Integer.class, Migration.class);
+                            mapBinder.addBinding(1)
+                                .toInstance(() -> Migration.MigrationResult.COMPLETED);
```
Comment (mbaechler) [inline server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java]: why ?

File: src/site/xdoc/server/config-cassandra.xml [hunk 4]
```diff
@@ -114,13 +114,13 @@
         <dt><strong>mailbox.blob.part.size</strong></dt>
         <dd>Optional. Defaults to 102400 (100KB).<br/> Controls the size of blob parts used to store messages.</dd>
         <dt><strong>migration.v1.v2.on.the.fly</strong></dt>
-        <dd>Optional. Defaults to false.<br/> Controls wether v1 to v2 migration should be run on the fly.</dd>
+        <dd>Only available on tag cassandra_migration_v1_to_v2. Optional. Defaults to false.<br/> Controls wether v1 to v2 migration should be run on the fly.</dd>
```
Comment (mbaechler) [inline src/site/xdoc/server/config-cassandra.xml]: wow, the document will soon be clutered with migration options at this pace. Can't we find a better solution ?

Comment (chibenwa) [inline src/site/xdoc/server/config-cassandra.xml]: Found: move them in the migration section, give details about each migration.

capture d ecran de 2017-08-18 09-43-14

**Comment by chibenwa** [comment]
```
Failed tests: 
  ReadOnlyUsersLDAPRepositoryTest.testShouldStillWorksAfterRestartingLDAP:148 expected:<[tru]e> but was:<[fals]e>
```

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged
